example/__tests__/App.js

```
@@ -1,15 +1,157 @@
+      const person = JSON.parse(currentPerson);
+
+      expect(person).toHaveProperty('name', 'Harry');
+      expect(person.characteristics).toHaveProperty('hair', 'red');
+      expect(person.characteristics).toHaveProperty('shoeSize', 40);
+    });
+  });
+
+  describe('Callback based', () => {
+    it('can read/write data to/from storage', done => {
+      const newData = Math.floor(Math.random() * 1000);
+
+      AsyncStorage.setItem('key', newData, function() {
+        AsyncStorage.getItem('key', function(_, value) {
+          expect(value).toBe(newData);
+          done();
+        }).catch(e => done.fail(e));
+      });
+    });
+    it('can clear storage', done => {
+      AsyncStorage.setItem('temp_key', Math.random() * 1000, () => {
+        AsyncStorage.getItem('temp_key', (_, currentValue) => {
+          expect(currentValue).not.toBeNull();
+          AsyncStorage.clear(() => {
+            AsyncStorage.getItem('temp_key', (_, value) => {
+              expect(value).toBeNull();
+              done();
+            }).catch(e => done.fail(e));
+          });
+        }).catch(e => done.fail(e));
+      });
+    });
+
+    it('can clear entries in storage', done => {
+      AsyncStorage.setItem('random1', Math.random() * 1000, () => {
+        AsyncStorage.setItem('random2', Math.random() * 1000, () => {
+          AsyncStorage.getItem('random1', (_, data1) => {
+            AsyncStorage.getItem('random2', (_, data2) => {
+              expect(data1).not.toBeNull();
+              expect(data2).not.toBeNull();
+
+              AsyncStorage.removeItem('random1', () => {
+                AsyncStorage.removeItem('random2', () => {
+                  AsyncStorage.getItem('random1', (_, value1) => {
+                    AsyncStorage.getItem('random2', (_, value2) => {
+                      expect(value1).toBeNull();
+                      expect(value2).toBeNull();
+                      done();
+                    }).catch(e => done.fail(e));
```

you're using callback approach, but here taking advantage of `getItem` returning a promise. You should use the first argument (named `_` instead of `err` or `error`) and check if it's not null, then call `done.fail`

---

The reason why I'm catching some of the block is that `error` thrown in callback methods are swallowed by Jest - meaning it'll get `timeout` singal, without specific reason why caused that.

Note that I only wrap blocks with assertions inside - I couldn't find nicer way to do `callbacks` and still get errors when assertion fail